src/analyze/analyze-security.c

```diff
@@ -2708,6 +2709,12 @@ static int offline_security_checks(
 
         if (profile) {
                 /* Ensure the temporary directory is in the search path, so that we can add drop-ins. */
+                if (!m->lookup_paths.temporary_dir) {
+                        r = mkdtemp_malloc("/tmp/systemd-temporary-XXXXXX", &m->lookup_paths.temporary_dir);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to create temporary directory: %m");
+                }
```

Comment:
hmm the LOOKUP_PATHS_TEMPORARY_GENERATED flag to lookup_paths_init() is already doing just this. are we not setting that here? shouldn't we do that?

i.e. instead of reaching down here, i think it would make sense for LookupPaths to already be initialized properly right-away?